src/patternfly/components/Nav/nav.scss

```diff
@@ -354,389 +235,275 @@
+  background-color: var(--pf-c-nav__link--BackgroundColor);
+
+  &::before {
+    position: absolute;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    content: "";
+    border-color: var(--pf-c-nav__link--before--BorderColor);
+    border-style: solid;
+    border-width: 0 0 var(--pf-c-nav__link--before--BorderBottomWidth) var(--pf-c-nav__link--before--BorderLeftWidth);
   }
 
   // States
-  .pf-m-current.pf-c-nav__link,
-  .pf-m-current > .pf-c-nav__link {
-    --pf-c-nav__list-link--FontWeight: var(--pf-c-nav__list-link--m-current--FontWeight);
-    --pf-c-nav__list-link--Color: var(--pf-c-nav__list-link--m-current--Color);
-
-    &::after {
-      background-color: var(--pf-c-nav__list-link--m-current--after--BackgroundColor);
-    }
+  &:hover {
+    color: var(--pf-c-nav__link--hover--Color);
+    background-color: var(--pf-c-nav__link--hover--BackgroundColor);
   }
-}
-
 
-// Separator --- remove this block at breaking change
-.pf-c-nav__separator {
-  height: var(--pf-c-nav__separator--Height);
-  margin-top: var(--pf-c-nav__separator--MarginTop);
-  margin-bottom: var(--pf-c-nav__separator--MarginBottom);
-  margin-left: var(--pf-c-nav__separator--MarginLeft);
-  background-color: var(--pf-c-nav__separator--BackgroundColor);
+  &:focus {
+    color: var(--pf-c-nav__link--focus--Color);
+    background-color: var(--pf-c-nav__link--focus--BackgroundColor);
+  }
 
-  .pf-c-nav__simple-list & {
-    --pf-c-nav__separator--MarginLeft: var(--pf-c-nav__simple-list__separator--MarginLeft);
+  &:active {
+    color: var(--pf-c-nav__link--active--Color);
+    background-color: var(--pf-c-nav__link--active--BackgroundColor);
   }
 
-  .pf-c-nav__item & {
-    --pf-c-nav__separator--MarginLeft: var(--pf-c-nav__simple-list--nested__separator--MarginLeft);
+  &.pf-m-current {
+    color: var(--pf-c-nav__link--m-current--Color);
+    background-color: var(--pf-c-nav__link--m-current--BackgroundColor);
   }
-}
 
-// Toggle caret
-.pf-c-nav__toggle {
-  flex: none;
-  padding-right: var(--pf-c-nav__list-toggle--PaddingRight);
-  padding-left: var(--pf-c-nav__list-toggle--PaddingLeft);
-  margin-left: auto;
-  font-size: var(--pf-c-nav__list-toggle--FontSize);
-  line-height: 1;
-  pointer-events: none;
-  border: 0;
+  &:disabled,
+  &.pf-m-disabled {
+    color: var(--pf-c-nav__link--disabled--Color);
+    pointer-events: none;
+    background-color: var(--pf-c-nav__link--disabled--BackgroundColor);
+  }
 
-  // Add transition to base
-  & > * {
-    transition: var(--pf-c-nav__list-toggle--Transition);
+  &,
+  &:hover,
+  &:focus,
+  &:active {
+    text-decoration: none;
   }
 }
 
-// Rotate the icon
-.pf-c-nav__item.pf-m-expanded .pf-c-nav__toggle > * {
-  transform: var(--pf-c-nav__item--m-expanded__toggle-icon--Transform);
+// Active border treatment
+.pf-c-nav__list,
+.pf-c-nav__simple-list {
+  .pf-c-nav__link.pf-m-current {
+    --pf-c-nav__link--before--BorderLeftWidth: var(--pf-c-nav__link--before--BorderWidth);
```

Comment:
I wonder if we should separate these out so they aren't tied to one another? If we style them differently and `--pf-c-nav__link--before--BorderWidth` doesn't control both, that's a breaking change.